### PR TITLE
Wallet self updating

### PR DESCRIPTION
**Pull request by ngvtuan**
##### Commits
- [x] Make wallet self-updating so it updates the same way Chrome and VScode do
- [x] Done for window version

##### FreedCamp
- [x] [Make wallet self-updating so it updates the same way Chrome and VScode do](https://freedcamp.com/view/2956183/tasks/38537599)

### CODE REVIEW

**Comment by br4in3x** [comment]
Perhaps we can update this to be a "Draft PR" if this is still in progress?

File: forge.config.js [hunk 1]
```diff
@@ -40,19 +43,32 @@ module.exports = {
...
+    //   identity: 'Developer ID Application: Felix Rieseberg (LT94ZKYDCJ)',
+    //   'hardened-runtime': true,
+    //   entitlements: 'static/entitlements.plist',
+    //   'entitlements-inherit': 'static/entitlements.plist',
+    //   'signature-flags': 'library',
+    // },
+    // osxNotarize: {
+    //   appleId: 'felix@felix.fun',
+    //   appleIdPassword: 'my-apple-id-password',
+    // },
```
Comment (br4in3x) [inline forge.config.js:56]: Do we need this? If this is going to be needed in the future, let's leave it

File: forge.config.js [hunk 2]
```diff
@@ -40,19 +43,32 @@ module.exports = {
...
+        return {
+          exe: `${package.name}.exe`,
+          setupIcon: './static/icons/icon.ico',
+          loadingGif: './static/icons/icon.gif',
+          iconUrl:
+            'https://raw.githubusercontent.com/pastelnetwork/pastel-electron-wallet/master/static/icons/icon.ico',
+          title: package.productName,
+          setupExe: `${package.productName} Setup - v${package.version}.exe`,
+          skipUpdateIcon: true,
+        }
```
Comment (br4in3x) [inline forge.config.js:71]: Could you explain what's the reason for this change?

File: src/features/Common.module.css [hunk 3]
```diff
@@ -0,0 +1,38 @@
+.highlight {
```
Comment (br4in3x) [inline src/features/Common.module.css:1]: Let's put it in the common folder

File: src/features/updateToast/UpdateToast.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    shell.openExternal(`https://github.com/${pkg.repoName}/releases`)
+  }
+
+  const handleUpdate = () => {
+    dispatch(closeUpdateToast())
+    ipcRenderer.send('restart_app')
+  }
+
+  if (!opened) {
+    return <></>
```
Comment (br4in3x) [inline src/features/updateToast/UpdateToast.tsx:25]:
```suggestion
    return null
```

File: src/features/updateToast/UpdateToastSlice.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    },
+    closeUpdateToast(state: IUpdateToastState) {
+      state.opened = false
+    },
+  },
+})
+
+export const updateToastReducer = updateToastSlice.reducer
+
+export const { openUpdateToast, closeUpdateToast } = updateToastSlice.actions
```
Comment (br4in3x) [inline src/features/updateToast/UpdateToastSlice.ts:27]: I think nothing else is going to use the `updateToast`, so it's okay to have it's state through `useState`

File: src/main.ts [hunk 6]
```diff
@@ -169,3 +174,67 @@ app.on('activate', () => {
...
+    }/${app.getVersion()}`
+
+    autoUpdater.setFeedURL({
+      url: feedURL,
+      serverType: 'default',
+    })
+    autoUpdater.checkForUpdates()
+    setInterval(() => {
+      autoUpdater.checkForUpdates()
+    }, 4 * 60 * 60 * 1000)
```
Comment (br4in3x) [inline src/main.ts:191]: nit: can we create a constant with the interval name, like 
```ts
const oneHour = 60 * 60
```

File: src/main.ts [hunk 7]
```diff
@@ -169,3 +174,67 @@ app.on('activate', () => {
...
+
+autoUpdater.on('error', err => {
+  console.log('error', { err })
+})
+
+const checkSignatureExists = () => {
+  if (os.platform() === 'darwin') {
+    const certExists = fs.existsSync(
+      path.join(__dirname, 'static', 'entitlements.plist'),
+    )
```
Comment (br4in3x) [inline src/main.ts:234]: This is not a certificate. This is an MacOS app permission manifest. Not sure if it's valid to use it for detecting signature

File: src/features/updateToast/UpdateToast.tsx (hunk 4 above)
Comment (ngvtuan) [reply src/features/updateToast/UpdateToast.tsx]: Updated

File: src/main.ts (hunk 6 above)
Comment (ngvtuan) [reply src/main.ts]: Updated

File: src/main.ts (hunk 7 above)
Comment (ngvtuan) [reply src/main.ts]: Yeah, I think we don't need it. Removed

File: src/features/Common.module.css (hunk 3 above)
Comment (ngvtuan) [reply src/features/Common.module.css]: Updated

File: forge.config.js (hunk 1 above)
Comment (ngvtuan) [reply forge.config.js]: Yes we need it for MAC. and I'm waiting for the Signing Code

**Review by br4in3x — APPROVED** [review]

**Review by br4in3x — APPROVED** [review]
